packages/broken-link-checker/src/run.js

```diff
@@ -15,9 +15,16 @@ const run = ({ server, port, config, checkerOptions }) => {
     html: (/* tree, robots, response, pageUrl, customData */) => {},
     junk: (/* link, customData */) => {},
     link: (link /* , customData */) => {
+      if (!link.internal) {
+        const { protocol } = new URL(link.url.original);
```

Comment:
I think you don't need to re-parse here - not tested it now but I remember `${link.url.parsed.protocol}` is available (`link.url.parsed` is already an URL object or at least like one).